packages/process/src/node/terminal-process.ts

```diff
@@ -226,6 +238,19 @@ export class TerminalProcess extends Process {
                 }
             });
 
+            /**
+             * Wait for all the streams to be closed, as well as for the process
+             * to exit.
+             */
+            Promise.all([onExit, onClose]).then(resolved => {
+                const { code, signal } = resolved[0];
+                if (signal === undefined || signal === 0) {
+                    this.emitOnClose(code, undefined);
+                } else {
+                    this.emitOnClose(undefined, signame(signal));
```

Comment:
`code` should not be passed here?

Reply:
A program only ever exits with a code or a signal, never both. The above code just ensure that we don't get "null-like values" (0 or `null`) but `undefined`.

Reply:
> A program only ever exits with a code or a signal, never both. 
bikeshedding: it's always a code, and `$? returns (128 + n)`, where `n` is the signal. so there might be no signal.

Reply:
@AlexTugarev ah indeed, my bad. For my defense let's say I aligned with Node's behavior: https://nodejs.org/api/child_process.html#child_process_event_exit ;)